server/apiserver/argoserver.go

```diff
@@ -300,17 +303,34 @@ func (as *argoServer) newHTTPServer(ctx context.Context, port int, artifactServe
 	mustRegisterGWHandler(workflowarchivepkg.RegisterArchivedWorkflowServiceHandlerFromEndpoint, ctx, gwmux, endpoint, dialOpts)
 	mustRegisterGWHandler(clusterwftemplatepkg.RegisterClusterWorkflowTemplateServiceHandlerFromEndpoint, ctx, gwmux, endpoint, dialOpts)
 
-	mux.HandleFunc("/api/", func(w http.ResponseWriter, r *http.Request) { webhookInterceptor(w, r, gwmux) })
+	mux.HandleFunc("/api/", configureCORS(func(w http.ResponseWriter, r *http.Request) { webhookInterceptor(w, r, gwmux) }, as.accessControlAllowOrigin))
 	mux.HandleFunc("/artifacts/", artifactServer.GetArtifact)
 	mux.HandleFunc("/artifacts-by-uid/", artifactServer.GetArtifactByUID)
 	mux.HandleFunc("/oauth2/redirect", as.oAuth2Service.HandleRedirect)
 	mux.HandleFunc("/oauth2/callback", as.oAuth2Service.HandleCallback)
 	mux.Handle("/metrics", promhttp.Handler())
 	// we only enable HTST if we are secure mode, otherwise you would never be able access the UI
-	mux.HandleFunc("/", static.NewFilesServer(as.baseHRef, as.tlsConfig != nil && as.hsts, as.xframeOptions).ServerFiles)
+	mux.HandleFunc("/", configureCORS(static.NewFilesServer(as.baseHRef, as.tlsConfig != nil && as.hsts, as.xframeOptions).ServerFiles, as.accessControlAllowOrigin))
 	return &httpServer
 }
 
+// configureCORS is a middleware for setting CORS headers
```

Comment:
Looks good. Any reason to not insert this next to the x-frames-options code?

Reply:
I just want this middleware close to the callers. Should I move it?

Reply:
Look for where we already add headers suck as X-Frame-Options

Reply:
Okay found it. Should I not use the middleware and just call `w.Header().Set()` in `static.go`?

Reply:
Moved it near the x-frame-options code, kindly check.

Reply:
@joyciep After this change, are the CORS configured also for `/api/` URI? If I am reading it right, it is only used for `static.NewFilesServer`, isn't it?